src/components/PublicStorageService.js

```diff
@@ -59,18 +65,35 @@ goog.provide('ga_publicstorage_service');
         //     returns the same adminId and the same file url
         // if id is an fileId --> fork the file
         //     returns new adminId and new file url
-        this.save = function(endPoint, publicEndPoint, id, content,
-            contentType) {
-          return $http.post(getServiceUrl(endPoint, id), content, {
+        this.save = function(endPoint, publicEndPoint, fileId,
+            adminId, content, contentType) {
+
+          // const method = (id === undefined) ? 'POST' : 'PUT'
+          var method = 'PUT'
+          const formData = new FormData();
+          const blob = new Blob([content], { name: 'kml',
+            filename: 'blob',
+            type: contentType})
+          formData.append('kml', blob);
+          formData.append('author', 'mf-geoadmin3');
+          if (fileId === undefined && adminId === undefined) {
+            method = 'POST'
+          } else {
+            formData.append('admin_id', adminId)
+          }
+          return $http({
+            method: method,
+            url: getServiceUrl(endPoint, fileId),
+            data: formData,
             headers: {
-              'Content-Type': contentType
+              'Content-Type': undefined
```

Comment:
Is it worth keeping?

Reply:
Yes, it's angularjs black magic. Without it, the default `application\json` will be sent, with it, `multipart\form-data`

Reply:
I did comment to explain why